src/services/ui/src/components/Routing/types.ts

```diff
@@ -0,0 +1,63 @@
+import type { UrlObject } from "url";
```

Comment:
This is where the sausage gets made.  I will follow up with comments/documentation when I understand it.